sycl/doc/extensions/GlobalWorkItems/SYCL_INTEL_global_work_items.asciidoc

```diff
@@ -0,0 +1,184 @@
+item<dimensions> this_item();
+----
+
+===== Add the table with free functions description
+
+|===
+a|
+[source,c++,multiptr,linenums]
+----
+template <int dimensions>
+item<dimensions> this_item();
+---- a|
+Returns the `item` instance representing the current work item in a `range`. The prerequisites:
+
+* `dimensions` value passed to `this_item` function shall match `dimensions` value passed to the `range` when the `parallel_for` is called
+
+* shall only be invoked within the `parallel_for` context with `range` as the argument
+
+If prerequisites are not fulfilled the behavior is undefined
+|===
+
+=== Changes in 4.10.1.5 (nd_item class)
+
+==== Changes in synopsis
+
+===== Add the following free function
+
+[source,c++,multiptr,linenums]
+----
+template <int dimensions>
+nd_item<dimensions> this_nd_item();
+----
+
+===== Add the table with free functions description
+
+|===
+a|
+[source,c++,multiptr,linenums]
+----
+template <int dimensions>
+nd_item<dimensions> this_nd_item();
+---- a|
+Returns the `nd_item` instance representing the current work item in a `nd_range`. The prerequisites:
+
+* `dimensions` value passed to `this_nd_item` function shall match `dimensions` value passed to the `nd_range` when the `parallel_for` is called
+
+* shall only be invoked within the `parallel_for` context with `nd_range` as the argument
+
+If prerequisites are not fulfilled the behavior is undefined
+|===
+
+=== Changes in 4.10.1.6 (h_item class)
+
+==== Changes in synopsis
+
+===== Add the following free function
+
+[source,c++,multiptr,linenums]
+----
+template <int dimensions>
+h_item<dimensions> this_h_item();
+----
+
+===== Add the table with free functions description
+
+|===
+a|
+[source,c++,multiptr,linenums]
+----
+template <int dimensions>
+h_item<dimensions> this_h_item();
+---- a|
+Returns the `h_item` instance representing the work item in a local `range` passed to a `parallel_for_work_item`. The prerequisites:
+
+* `dimensions` value passed to `this_h_item` function shall match `dimensions` value passed to the `range` when the `parallel_for_work_group` is called
+
+* shall only be called from within the`parallel_for_work_item` context
+
+If prerequisites are not fulfilled the behavior is undefined
+|===
+
+=== Changes in 4.10.1.7 (group class)
+
+==== Changes in synopsis
+
+===== Add the following free function
+
+[source,c++,multiptr,linenums]
+----
+template <int dimensions>
+group<dimensions> this_group();
+----
+
+===== Add the table with free functions description
+
+|===
+a|
+[source,c++,multiptr,linenums]
+----
+template <int dimensions>
+group<dimensions> this_group();
+---- a|
+Returns the `group` instance representing the particular work-group within the parallel execution. The prerequisites:
+
+* `dimensions` value passed to `this_group` function shall match `dimensions` value passed to the `range` when the `parallel_for_work_group` is called
+
+* shall only be called from within the`parallel_for_work_group` context
```

Comment:
I think `this_group` should work in `nd_range` kernels as well.  SYCL 2020 provisional removed several functions from `nd_item` (e.g. `barrier`, `mem_fence`) and generally we're trying to move in the direction of recommending that users pass groups around instead of items.  

Being able to jump straight to the `group` with these queries could also be a useful shorthand.  For example, compare the following usages of the group algorithms:
```cpp
sycl::reduce(this_nd_item<2>().get_group(), x, plus<>());
sycl::reduce(this_group<2>(), x, plus<>());
```

Reply:
Done. And also permit the call of this function for `parallel_for` with `nd_range` only. Removed the permission to call it for `parallel_for_work_group` as @rolandschulz suggested